### PR TITLE
add Support for ByteArray in Core and Client 🚀

### PR DESCRIPTION
**Pull request by RohrerF**
Adds ByteArrayType to Core and Client

Tasks:

- [x] Implement in Core
- [x] Tests ByteArrayType
- [x] Implement in Client
- [x] Test in Client

### CODE REVIEW

File: src/Core/Types/Types/Scalars/Scalars.cs [hunk 1]
```diff
@@ -82,6 +84,8 @@ public static class Scalars
                     typeof(NameType), TypeContext.None) },
                 { ScalarNames.PaginationAmount, new ClrTypeReference(
                     typeof(PaginationAmountType), TypeContext.None) },
+                { ScalarNames.ByteArray, new ClrTypeReference(
+                    typeof(ByteArrayType), TypeContext.None) },
            };
 
         private static readonly Dictionary<Type, ValueKind> _scalarKinds =
```
Comment (RohrerF) [inline src/Core/Types/Types/Scalars/Scalars.cs]: @michaelstaib not sure if byte[] needs to be added here.

**Comment by RohrerF** [comment]
@michaelstaib this PR is ready for review

File: src/Core/Types/Types/Scalars/Scalars.cs (hunk 1 above)
Comment (michaelstaib) [reply src/Core/Types/Types/Scalars/Scalars.cs]: Adding it here will add this as a default resolution meaning that the schema will now resolver byte arrays automatically to the type `ByteArray`. So, this should be fine.

**Comment by michaelstaib** [comment]
This one is now in the merge queue :) should go in today ...

**Pull request merged**